### PR TITLE
#698 NOTES - SHELTER-SHELTER INTERVIEW  REMOVE SELF PAY 

### PR DESCRIPTION
**Pull request by MiKaylaH22**
For the shelter interview script:
REMOVE:
- [x] Mention of self-pay -close issue https://github.com/Hennepin-County/MAXIS-scripts/issues/698
- [x] Amt issued to EBT Line and replace with PN $70 PER HH member
- [x] Social Worker
- [x] 100% of cash benefits
- [x] 1st two asterisk at the bottom
ADD:  

- [x] Header of dialog - SHELTER INTERVIEW  
- [x] Case note header - Shelter Interview:   Family in Shelter   
- [x] Read CASE/CURR for current cash program   
- [x] Possible HH dropdown selection?  
- [x] Add Partners selection list    
- [x] On the first script we want to add Client answered "yes/no" to all health screening questions.   
- [x] add homelessness verified.  
INDIVIDUAL SCRIPTS TO ADDRESS WITH SELF PAY
- [x] Cash Cutoff retire
- [x]  [shelter/shelter-selfpay.vbs](https://github.com/Hennepin-County/MAXIS- retire 
- [x]  [shelter/shelter-money-mismanagement.vbs](https://github.com/Hennepin-County/MAXIS- remove mention
- [x]  [shelter/shelter-revoucher.vbs](https://github.com/Hennepin-County/MAXIS- remove mention
- [x]  [shelter/shelter-main-menu.vbs](https://github.com/Hennepin-County/MAXIS- remove mention

### CODE REVIEW

File: shelter/shelter-shelter-interview.vbs [hunk 1]
```diff
@@ -101,23 +161,29 @@ EMWriteScreen MAXIS_case_number, 18, 43
...
-If cash_type = "DWP" then
-    Call write_variable_in_CASE_NOTE("* DWP families are not eligible for the one-time only personal needs and medical co-pays")
-ELSE
-    Call write_variable_in_CASE_NOTE(" Except $" & one_time_issuance & " to EBT for one-time only (10%) for personal needs. $20 for medical co-pays.")
-END IF
-Call write_variable_in_CASE_NOTE("---")
-Call write_bullet_and_variable_in_CASE_NOTE("Other income", other_income)
-Call write_bullet_and_variable_in_CASE_NOTE("Money mismanagement", money_mismanagement)
-Call write_bullet_and_variable_in_CASE_NOTE("Reason for homelessness", reason_homeless)
+CALL write_variable_in_CASE_NOTE("### Family in Shelter Interview with M" &  MEMB_number & " ####")
```
Comment (C-Love) [inline shelter/shelter-shelter-interview.vbs:171]: Is this only for family cases? Was this script for anything other than self-pay issues?

**Review by C-Love — CHANGES_REQUESTED** [review]
Some additional items to address:

Changelog needs to be addressed.
Closing project documentation needs to be added.

In shelter-revoucher
  - statistics needs to be addressed 
  - goals_number on 166 and 188 should not be the same variable, they will overwrite each other.
  - the dialogs on 169 and 190 should be named Dialog1
  - shelter_dates on 125 is never called
  - 182 and 203 are using stopscript

shelter-interview
The purpose of this script is a bit confusing, so it is a bit hard to assess the functionality. It seems that some of the purpose of the script has changed with this functionality update. It isn't just removing the self pay portions.
I hit some bugs when running the script
 - determine_program_and_case_status_from_CASE_CURR had the wrong number of parameters
 - when I pressed the shelter button, the dialog did not reappear

**Comment by MiKaylaH22** [comment]
- [ ] > Changelog needs to be addressed. 

- [ ] Closing project documentation needs to be added. 

 
> In shelter-revoucher

- [ ] > * statistics needs to be addressed - recommend this be out of scope to hit deadline 
- [ ] > * goals_number on 166 and 188 should not be the same variable, they will overwrite each other.
- [ ] > * the dialogs on 169 and 190 should be named Dialog1
- [ ] > * shelter_dates on 125 is never called
- [ ] > * 182 and 203 are using stopscript

> 
> shelter-interview The purpose of this script is a bit confusing, so it is a bit hard to assess the functionality. It seems that some of the purpose of the script has changed with this functionality update. It isn't just removing the self pay portions. I hit some bugs when running the script
 

- [ ] > * determine_program_and_case_status_from_CASE_CURR had the wrong number of parameters

- [ ] > * when I pressed the shelter button, the dialog did not reappear

File: shelter/shelter-money-mismanagement.vbs [hunk 2]
```diff
@@ -53,54 +53,50 @@ changelog_display
...
-  EditBox 245, 65, 35, 15, to_date
-  EditBox 50, 85, 240, 15, Comments_notes
-  CheckBox 5, 110, 100, 10, "Income no longer available", income_checkbox
+  ComboBox 160, 5, 60, 15, "Phone call"+chr(9)+"Voicemail"+chr(9)+"Email"+chr(9)+"Office visit"+chr(9)+"Letter", contact_type
+  DropListBox 225, 5, 45, 10, "to"+chr(9)+"from", contact_direction
+  ComboBox 275, 5, 65, 15, "Client"+chr(9)+"Other HH Memb"+chr(9)+"AREP", who_contacted
+  EditBox 100, 25, 45, 15, date_requested
+  EditBox 185, 25, 85, 15, phone_number
+  DropListBox 50, 45, 145, 15, "1st Instance of Money Mismanagement"+chr(9)+"2nd Instance of Money Mismanagement"+chr(9)+"Grant Management", Occurrence_droplist
+  CheckBox 210, 45, 100, 10, "Income no longer available", income_checkbox
```
[the remark this answers is not in the document]
Comment (MiKaylaH22) [reply shelter/shelter-money-mismanagement.vbs:69]: removed

File: shelter/shelter-money-mismanagement.vbs [hunk 3]
```diff
@@ -53,54 +53,50 @@ changelog_display
...
 EMWriteScreen CM_mo, 20, 43	'entering current footer month/year
 EMWriteScreen CM_yr, 20, 46
-when_contact_was_made = date
-date_requested = date & ""
-income_checkbox = checked
-'1st Instance of Money Mismanagement = occurrence_MM1--------SAVE for ENHANCEMNT
-'2nd Instance of Money Mismanagement = occurrence_MM2
-'Grant Management = occurrence_MM3
 
 months_variable = CM_mo
```
[the remark this answers is not in the document]
Comment (MiKaylaH22) [reply shelter/shelter-money-mismanagement.vbs:101]: this has been addressed and is only used when in grant management

File: shelter/shelter-revoucher.vbs [hunk 4]
```diff
@@ -67,8 +67,8 @@ DO
 		err_msg = ""
 		Dialog Dialog1
         cancel_without_confirmation
-		IF len(MAXIS_case_number) > 8 or IsNumeric(MAXIS_case_number) = False THEN err_msg = err_msg & vbNewLine & "* Enter a valid case number."
-		IF revoucher_option = "Select one..." then err_msg = err_msg & vbNewLine & "* Select a revoucher option."
+		Call validate_MAXIS_case_number(err_msg, "*")
+		IF revoucher_option = "Select one..." then err_msg = err_msg & vbNewLine & "* Please select a revoucher option."
 		IF err_msg <> "" THEN MsgBox "*** NOTICE!!! ***" & vbNewLine & err_msg & vbNewLine
 	LOOP UNTIL err_msg = ""
  Call check_for_password(are_we_passworded_out)
```
[the remark this answers is not in the document]
Comment (MiKaylaH22) [reply shelter/shelter-revoucher.vbs:74]: i may have fixed this and not known it but it looks right now

File: shelter/shelter-shelter-interview.vbs (hunk 1 above)
Comment (MiKaylaH22) [reply shelter/shelter-shelter-interview.vbs:171]: Interviews are done on all family cases looking to shelter

**Comment by MiKaylaH22** [comment]
after review with Casey and updates made to worker signature merge request was made. Change Reported is not in scope of the project and was pulled back from this commit.

**Pull request merged**